flight/targets/dtfc/fw/pios_board.c

```diff
@@ -255,17 +255,10 @@ void PIOS_Board_Init(void)
 #endif
 
 #if defined(PIOS_INCLUDE_ADC)
-    uintptr_t internal_adc_id;
-	if (PIOS_INTERNAL_ADC_Init(&internal_adc_id, &internal_adc_cfg) < 0)
-		PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_ADC);
-	if (PIOS_ADC_Init(&pios_internal_adc_id, &pios_internal_adc_driver, internal_adc_id) < 0)
-		PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_ADC);
-
-	// Enable battery monitor module.
-	ModuleSettingsData modulesettings;
-	ModuleSettingsGet(&modulesettings);
-	modulesettings.AdminState[MODULESETTINGS_ADMINSTATE_BATTERY] = MODULESETTINGS_ADMINSTATE_ENABLED;
-	ModuleSettingsSet(&modulesettings);
+	uintptr_t unused_adc;
```

Comment:
If we're going to have the var, we may as well call it what it actually is right (i.e. internal_adc_id)? Otherwise, just change the API to accept+handle NULL and eliminate it.

Reply:
I don't know what to call these.  We have the actual internal_adc internal adc device handle, and the wrapped PIOS_ADC device handle for the internal ADC.  Both of which are mostly unused.  I can cast them to void, which seems like the lesser of evils.

Reply:
Oh, it's not a retval.

Reply:
So yah, the whole reason I did this was we had both pios_internal_adc_id which was the actual internal adc id device, and the wrapped version of the same, which was internal_adc_id.  If anything, the thing you'd want to have broader scope / the pios name is the wrapped version, so the names are swapped.  But I dare not swap just the F3 targets, because all the F4 targets would then be different.

Reply:
So, this is the one you actually want to use with PIOS_ADC_* (which is the interface everybody should use other than PIOS_INTERNAL_ADC_Init). I just had a quick look and found a bug:
https://github.com/d-ronin/dRonin/blob/cd7a75d880e031c72106747d779fb352bf645203/flight/Modules/System/systemmod.c#L761
That attempts to use a pios_internal_adc_dev as a pios_adc_dev.

Reply:
That code's already removed in this PR because no targets provided configuration for it.

Reply:
(it's also F4 specific; if we want to bring it back, we should have each pios_internal_adc driver provide an implementation rather than do target-specific configuration)